pytests/fts/random_query_generator/rand_query_gen.py

```diff
@@ -568,6 +568,191 @@ def construct_fuzzy_query(self):
 
         return fts_query, es_query
 
+    def generate_random_point():
+        lat = -90 + random.random()*180
+        long = -90 + random.random()*180
+        return lat,long
+
+    def generate_random_polygon():
+        vertices = random.randint(4,10)
+        lat_base = random.random() + random.randint(1,10)*6
+        long_base = random.random() + random.randint(1,8)*10
+        sign = -1.0
+        step = 0.95
+        coord = []
+        coord.append([long_base, lat_base])
+        for i in range(1,vertices+1):
+               lat_delta = sign * step
+               lat = coord[i-1][1] + lat_delta + random.randint(1,4)
+               long = long_base + 1.5   
+               coord.append([long, lat])
+               sign *= -1
+        coord.append([long_base, lat_base])
+        coord = [coord]
+        return coord
+
+    def generate_random_linestring():
+        lat1,lon1 = FTSESQueryGenerator.generate_random_point() 
+        lat2,lon2 = FTSESQueryGenerator.generate_random_point()
+        coord =[[lat1,lon1],[lat2,lon2]]
+        return coord
+
+    def generate_random_multipoint():
+        coord = []
+        n = random.randint(4,10)
+        for v in range(1,n+1):
+                lat,lon = FTSESQueryGenerator.generate_random_point() 
+                coord.append([lat,lon])
+        return coord
+    
+    def generate_random_multilinestring():
+        coord = []
+        n = random.randint(4,10)
+        for v in range(1,n+1):
+                line = FTSESQueryGenerator.generate_random_linestring() 
+                coord.append(line) 
+        return coord
+
+    def generate_random_multipolygon():
+         coord = []
+         n = random.randint(4,10)
+         for v in range(1,n+1):
+             coord.append(FTSESQueryGenerator.generate_random_polygon())
+         return coord
+
+    def generate_random_circle():
+        lat,lon = FTSESQueryGenerator.generate_random_point() 
+        coord = [lat,lon]
+        radius = str(random.randint(500,20000)) + "m"
+        return coord,radius
+
+    def generate_random_envelope():
+        lat1,lon1 = FTSESQueryGenerator.generate_random_point() 
+        diff = random.random() * random.randint(1,10)
+        lat2 = lat1 + diff
+        lon2 = lon1 - diff
+        coord = [[lat1,lon1],[lat2,lon2]]
+        return coord
+
+    @staticmethod
+    def construct_geo_shape_point_query():    
+        shapes = ['point','linestring','polygon','multipoint','multilinestring','multipolygon','circle',
+        'envelope','geometrycollection']
+        shape = random.choice(shapes)
+
+        fts_query = dict()
+        es_query = dict()
+        coord = []
+        radius = "1500m"
+        coll_shapes = []
+
+        if shape == 'point':
+            relations = ['intersects','within','contains']
```

Comment:
why the relations are repeatedly declared in every block?

Reply:
I have declared them separately for each block since some relations are not applicable for some shapes('within' is not applicable for linestrings/multilinestrings).